EasyLayout.Forms/EasyLayout.cs

```diff
@@ -567,112 +548,52 @@ private static int ConstantValue(Expression expr)
 
         private static bool IsConstant(Expression expr)
         {
-            if (expr.NodeType == ExpressionType.Constant)
+            switch (expr)
             {
-                return true;
-            }
-
-            if (expr.NodeType == ExpressionType.MemberAccess)
-            {
-                var mexpr = (MemberExpression)expr;
-                // todo: Does this even work, where the heck is MemberType?
-                if (mexpr.NodeType == ExpressionType.Constant)
-                {
+                case ConstantExpression _:
                     return true;
-                }
-                return false;
-            }
-
-            if (expr.NodeType == ExpressionType.Convert)
-            {
-                var cexpr = (UnaryExpression)expr;
-                return IsConstant(cexpr.Operand);
-            }
-
-            var methodCall = expr as MethodCallExpression;
-            if (methodCall != null)
-            {
-                return methodCall.Method.Name == nameof(ToConst);
+                case MemberExpression mexpr:
+                    // todo: Does this even work, where the heck is MemberType?
+                    return mexpr.NodeType == ExpressionType.Constant;
+                case UnaryExpression cexpr:
+                    return IsConstant(cexpr.Operand);
+                case MethodCallExpression methodCall:
+                    return methodCall.Method.Name == nameof(ToConst);
+                default:
+                    return false;
             }
-
-            return false;
         }
 
         private static LeftExpression ParseLeftExpression(Expression expr)
         {
-            Position? position = null;
-            MemberExpression viewExpr = null;
-
             var fExpr = expr as MethodCallExpression;
-            if (fExpr != null)
-            {
-                position = GetPosition(fExpr);
-                viewExpr = fExpr.Arguments.FirstOrDefault() as MemberExpression;
-            }
-
-            if (position == null)
-            {
-                var memExpr = expr as MemberExpression;
-                if (memExpr == null)
-                {
-                    throw new NotSupportedException("Left hand side of a relation must be a member expression, instead it is " + expr);
```

Comment:
Between this and the line below it I think we may have lost a number of useful error message to the user.  Particularly with a new framework that has a lot of potential to get wrong, I think it's important to have the most useful error messages possible.  Can we get this back?  I feel like what we need are some unit tests to verify the old functionality.  Something like:

public void GivenWrongMethodName_WhenConstrainLayout_ThenNiceErrorMessage() {
    Assert.ThrowsException<NotSupportedException>(relativeLayout.ConstrainLayout(() => _view1.Frame.Right == _view2.Frame.Left), "Left hand side of a relation must be a member expression, instead it is Frame")
}